taca_ngi_pipeline/deliver/deliver_mosler.py

```diff
@@ -0,0 +1,292 @@
+        """
+        return db.update_project(db.dbcon(), self.projectid, delivery_status=status)
+
+    def deliver_project(self):
+        """ Deliver all samples in a project to mosler
+            
+            :returns: True if all samples were delivered successfully, False if
+                any sample was not properly delivered or ready to be delivered
+        """
+        try:
+            logger.info("Delivering {} to {}".format(
+                str(self), self.expand_path(self.moslerdeliverypath)))
+            if self.get_delivery_status() == 'DELIVERED' \
+                    and not self.force:
+                logger.info("{} has already been delivered".format(str(self)))
+                return True
+            # right now, don't catch any errors since we're assuming any thrown 
+            # errors needs to be handled by manual intervention
+            status = True
+            # Open sftp session with mosler, in this way multiple tranfer will be possible
+            try:
+                transport=paramiko.Transport(self.moslersftpserver)
+                password = getpass.getpass(prompt='Mosler Password for user {}:'.format(self.moslersftpserver_user))
+                transport.connect(username = self.moslersftpserver_user, password = password)
+            except Exception as e:
+                print 'Caught exception: {}: {}'.format(e.__class__, e)
+                raise
+
+            #memorize all samples that needs to be delivered
+            samples_to_deliver = [sentry['sampleid'] for sentry in db.project_sample_entries(
+                    db.dbcon(), self.projectid).get('samples', [])]
+            
+            # run multiple threads and store return functions
+            # http://stackoverflow.com/questions/6893968/how-to-get-the-return-value-from-a-thread-in-python
+            threads = [None] * len(samples_to_deliver)
+            results = [None] * len(samples_to_deliver)
+            thread  = 0
+            while len(samples_to_deliver) > 0:
+                # check how many samples there are in mosler sftp server
+                # create an sftp client for each sample (only one put can be done in one client. This was needed for the threaded version)
+                sftp_client = transport.open_sftp_client()
+                # move to the delivery directory in the sftp
+                sftp_client.chdir(self.expand_path(self.moslerdeliverypath))
+                # Count number of files
+                num_tar_files_mosler = len(sftp_client.listdir('.'))
+                # if the the number of remote tar files is higher the the maximum number of deliveries wait
+                if num_tar_files_mosler >= self.moslersftpmaxfiles:
+                    logger.info("More than {} files in Mosler sftp server".format(self.moslersftpmaxfiles))
+                    print "more than {} files in Mosler".format(self.moslersftpmaxfiles)
```

Comment:
Even though the `logger` is slightly misconfigured so that it doesn't output to stdout, I think it's better to fix that issue rather than duplicating the printing of messages.